pyfr/plugins/tavg.py

```diff
@@ -351,3 +357,192 @@ def __call__(self, intg):
                     self.tstart_acc = intg.tcurr
 
                 self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(BaseCLIPlugin):
+    name = 'tavg'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    @classmethod
+    def add_cli(cls, parser):
+        cmd = cls.name
+        cp = parser.add_parser(cmd, help=f'{cmd} --help')
+        sp = cp.add_subparsers(dest='sub_cmd')
+
+        # Merge command
+        scmd = 'merge'
+        ssp = sp.add_parser(scmd, help=f'{cmd} {scmd} --help')
+        ssp.set_defaults(process=TavgCLIPlugin.merge_cli)
+        ssp.add_argument('solns', nargs='*', help='Averages to merge')
+        ssp.add_argument('output', help='Output file name')
+
+    @cli_external
+    def merge_cli(self):
+
+        # Open all the solition files
+        self._prepare_files(self.args.solns)
+
+        # Build up which files are needed and if std can be created
+        active = self._active_files()
+        merged_times = [(r.start, r.end) for r, f, s in active]
+        avg_time = sum([t[1] - t[0] for t in merged_times])
+        self.std_all = all([s == 'all' for r, f, s in active])
+
+        # Initialise things needed for the merge
+        r0, f0, s0 = active[0]
+        self._init_tavg_merge(*f0)
+
+        # Check all files are compatible for merge
+        self._compatibility_merge(active)
+
+        with h5py.File(self.args.output, 'w') as out:
```

Comment:
f or outf are probably better names to make it clear they are file objects rather than arrays/dicts.